CuriousKatie/Resources/Match.swift

```diff
@@ -0,0 +1,138 @@
+    /// - Parameter partner: Potential partner
+    /// - Returns: Matching score
+    private static func calculateScore(between seeker: Person, and partner: Person) -> Int {
+        let seekersInterests = seeker.interests.map { $0.name }
+        let partnersInterests = partner.interests.map { $0.name }
+        let differentInterests = seekersInterests.difference(from: partnersInterests)
+        return differentInterests.count
+    }
+    
+    /// Finding the best matching pairs combination from an array of People.
+    /// The best pairs are determined by how high is the combined score of all the possible matches in the array of People.
+    /// The array with the highest combined score is the one with the best matches.
+    /// - Parameter participants: Array of People participating
+    /// - Returns: Best matching pairs
+    static func bestMatches(from participants: [Person]) -> [Match] {
+        var maxScore = 0
+        var bestPairs = [Match]()
+        let allPossiblePairs = Match.matchingPairs(from: Match.pairsCombinations(from: participants, taking: 2))
+        
+        for index in 0..<allPossiblePairs.count {
+            var pairsScore = 0
+            var iterations = index
+            var possiblePairs = [Match]()
+            var pairs = allPossiblePairs
+            
+            while pairs.count != 0 {
+                iterations = checkIndex(pairs.count, index)
+                let match = pairs[iterations]
+                possiblePairs.append(match)
+                pairs.removeAll { $0.seeker == match.seeker }
+                pairs.removeAll { $0.seeker == match.partner }
+                pairs.removeAll { $0.partner == match.partner }
+                pairs.removeAll { $0.partner == match.seeker }
+            }
+            
+            pairsScore = Match.calculateMatchesScore(from: possiblePairs)
+            if maxScore < pairsScore {
+                maxScore = pairsScore
+                bestPairs = possiblePairs
+            }
+        }
+        
+        print("This group score is \(maxScore)/\(bestPairs.count * 20).")
+        return bestPairs
+    }
+    
+    /// Check if the index is out of the array.
+    /// - Parameter size: Array size
+    /// - Parameter index: Array index
+    /// - Returns: Either the index or in case the index is outside the array size - the last index
+    private static func checkIndex(_ size: Int, _ index: Int) -> Int {
+        var iterations = index
+        if size <= index {
+            iterations = size - 1
+        }
+        return iterations
+    }
+    
+    /// Calculating the score of a possible pairs combination.
+    /// - Parameter matches: Array of Matches
+    /// - Returns: Sum of the scores of all Matches
+    private static func calculateMatchesScore(from matches: [Match]) -> Int {
+        var score = 0
+        
+        for match in matches {
+            score += match.score
+        }
+        
+        return score
```

Comment:
FYI, Swift developer are very fan of these unfriendly (at first) array utilities: https://useyourloaf.com/blog/swift-guide-to-map-filter-reduce/
You can use one of them to reduce your code to one line.